charts/cryostat/templates/NOTES.txt

```diff
@@ -36,10 +36,10 @@
 
 {{ $listNum }}. Forward local port to the application's pod:
   ```
-  kubectl -n {{ .Release.Namespace }} wait --for=condition=available --timeout=60s deploy/{{ include "cryostat.fullname" . }}
+  kubectl -n {{ .Release.Namespace }} wait --for=condition=available --timeout=60s deploy/{{ include "cryostat.deploymentName" . }}
 
-  export POD_NAME=$(kubectl get pods -n {{ .Release.Namespace }} -l "app.kubernetes.io/name={{ include "cryostat.name" . }},app.kubernetes.io/instance={{ .Release.Name }}" --sort-by=.metadata.creationTimestamp -o jsonpath="{.items[-1:].metadata.name}")
-  export CONTAINER_PORT=$(kubectl get pod -n {{ .Release.Namespace }} $POD_NAME -o jsonpath="{.spec.containers[0].ports[0].containerPort}")
+  export POD_NAME=$(kubectl get pods -n {{ .Release.Namespace }} -l "app.kubernetes.io/name={{ include "cryostat.name" . }},app.kubernetes.io/instance={{ .Release.Name }},app.kubernetes.io/component=cryostat" --sort-by=.metadata.creationTimestamp -o jsonpath="{.items[-1:].metadata.name}")
+  export CONTAINER_PORT=$(kubectl get pod -n {{ .Release.Namespace }} $POD_NAME -o jsonpath='{.spec.containers[?(@.name=="{{ include "cryostat.fullname" . }}-authproxy")].ports[0].containerPort}')
```

Comment:
Could we give a name to the port of the oauth-proxy container that we can reference here? We

```
  ports:
    - containerPort: 4180
      name: http
      protocol: TCP
```

```
kubectl -n {{ .Release.Namespace }} port-forward $POD_NAME "8080:http"
```

Reply:
> Using default chart values, I got a white screen when trying to view a recording in Grafana. Don't see any errors though, so not sure what's going on.

> Hmm, looks like the port-forward/proxying isn't working properly. The browser goes to /grafana/, but the response it receives back is the index.html of cryostat-web, not Grafana. I've been testing installation on OpenShift with the optional features enabled (helm install cryostat --set authentication.openshift.enabled=true --set core.route.enabled=true ./charts/cryostat/) and the Grafana setup works fine there still. I'll take a closer look at why it's broken with the defaults/port-forward/oauth2-proxy.

I was able to reproduce this issue when using a release name that is **not** `cryostat`: `helm install my-chart charts/cryostat/`

I think the name of proxy container in JSON query is incorrect. It should be:

https://github.com/cryostatio/cryostat-helm/blob/1d1cef8ace7aba9966037b902544b3bd4ef3415f/charts/cryostat/templates/_oauth2Proxy.tpl#L5

In this case, the env var `CONTAINER_PORT` is empty and the cryostat port `8181` is somehow picked up instead. This is not always though as sometimes the correct port is picked. This might be cause of the original issue with white screen.

**Screenshots**

![image](https://github.com/user-attachments/assets/56f696a1-7366-4f50-812e-e5c36fb93d68)